common/lib/client/auth.js

```diff
@@ -97,7 +108,33 @@ var Auth = (function() {
 			Logger.logAction(Logger.LOG_ERROR, 'Auth()', msg);
 			throw new Error(msg);
 		}
-	}
+	};
+
+	/**
+	 * Update the authentication options for this client;
+	 * this will also update these options on the realtime connection
+	 * so that they take effect immediately
+	 * @param options
+	 */
+	Auth.prototype.setOptions = function(options) {
+		this._setOptions(options);
+		if(this.client instanceof Realtime) {
+			this.client.connection.connectionManager.onAuthUpdated();
+		}
+	};
+
+	Auth.prototype.setClientId = function(clientId) {
+		if(clientId !== '*') {
+			if(this.clientId && this.clientId !== clientId) {
+				var msg = 'Unable to set incompatible clientId';
```

Comment:
> The library allows it, so you would get a failed connection instead of an exception which means the connection isn't touched.

What do yo mean the connection isn't touched?

> Given this is a temporary situation I'm not sure it really makes a big difference.

I agree, but assuming we allow anonymous -> identified clients in realtime in the near future, it would be nice to know it will just work without any further changes here